### ISSUE TITLE
clean up bufrlib.pyf

### ISSUE THREAD

**Issue by jbathegit** [NOAA-EMC/NCEPLIBS-bufr#220]
There are a lot of declarations in python/_bufrlib.pyf that aren't needed, and which in fact are downright misleading given that most of the referenced arrays are now dynamically allocated, rather than being fixed sizes at compile time.  Furthermore, entries are only needed in this file for subprograms which are actually called by the python codes and utilities, and not for every last subprogram in the library.

In short, this python/_bufrlib.pyf file could use a good clean-up.
